source/val/BasicBlock.h

```diff
@@ -118,29 +154,56 @@ class BasicBlock {
 
    private:
     const BasicBlock* current_;
+    std::function<const BasicBlock*(const BasicBlock*)> dom_func_;
   };
 
-  /// Returns an iterator which points to the current block
+  /// Returns an dominator iterator which points to the current block
   const DominatorIterator dom_begin() const;
+
+  /// Returns an dominator iterator which points to the current block
   DominatorIterator dom_begin();
 
-  /// Returns an iterator which points to one element past the first block
+  /// Returns an dominator iterator which points to one element past the first
```

Comment:
"past the first" is right in the dominator case but not right in the post-dominator case.  It's really the one element past the last valid block (in either direction).

Typo "an" -> "a"